### PR TITLE
Add a new chef_infra alias to the chef_zero provisioner (Recreating PR 1730)

### PR DESCRIPTION
**Pull request by i5pranay93**
# Description

We are renaming provisioner chef_zero to chef_infra and making it backward compatible,  also replacing chef_infra as the default provisioner from chef_solo.

## Issues Resolved

fixes https://github.com/test-kitchen/test-kitchen/issues/1708
fixes https://github.com/test-kitchen/test-kitchen/issues/1723
Recreating- https://github.com/test-kitchen/test-kitchen/pull/1730

## Check List

- [ ] All tests pass. See TESTING.md for details.
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable.

### CODE REVIEW

File: lib/kitchen/provisioner/chef_infra.rb [hunk 1]
```diff
@@ -0,0 +1,167 @@
+#
+# Author:: Fletcher Nichol (<fnichol@nichol.ca>)
+#
+# Copyright (C) 2013, Fletcher Nichol
```
Comment (marcparadise) [inline lib/kitchen/provisioner/chef_infra.rb:4]: We should add an additional copyright line here, `Copyright (C) Chef Software Inc.`

File: spec/kitchen/provisioner_spec.rb [hunk 2]
```diff
@@ -47,6 +47,21 @@ class Coolbeans < Kitchen::Provisioner::Base
       provisioner.must_be_kind_of Kitchen::Provisioner::Coolbeans
     end
 
+    it "returns a provisioner object of the chef infra for chef zero" do
+      provisioner = Kitchen::Provisioner.for_plugin("chef_zero", {})
+      provisioner.must_be_kind_of Kitchen::Provisioner::ChefInfra
+    end
+
+    it "returns a provisioner object of the chef infra for chef_infra" do
+      provisioner = Kitchen::Provisioner.for_plugin("chef_zero", {})
```
Comment (marcparadise) [inline spec/kitchen/provisioner_spec.rb:56]: Description and plugin don't match - it says 'chef_infra' is given but we're giving 'chef_zero'.

**Review by marcparadise — CHANGES_REQUESTED** [review]
This looks good to me, thanks for re-working this to clear up the history issues.  There are two minor corrections inline, after which I think this is good to go.

**Comment by marcparadise** [comment]
There also seems to be somethin ghappening with the history here. Maybe we could  pair up tomorrow morning (EDT) to resolve?

**Comment by i5pranay93** [comment]
Closing due to messed up commit history

**Pull request closed without merging**